src/OpenTracing/Exceptions/InvalidReferenceArgument.php

```diff
@@ -0,0 +1,21 @@
+<?php
+
+namespace OpenTracing\Exceptions;
+
+use InvalidArgumentException;
+
+final class InvalidReferenceArgument extends InvalidArgumentException
+{
+    public static function emptyType()
```

Comment:
Conceptually the exception is thrown due to and invalid argument in the reference (therefore `InvalidReferenceArgument` but the reason could be because of an **empty type** or an **invalid context** but the exceptional case is the same. Regarding the naming for the methods, I do believe those factories should express the reason behind the exception and I cannot think of how to achieve that with imperative verbs since they are not performing an action.

Additionaly I would not use `InvalidArgumentException` as it is too general and I think we should always return domain exceptions. More so, if I want to do an exception catching somewhere and I have another method throwing the same exception it will be painfull to decide the behaviour based on the message.